akka-http-core/src/main/scala/akka/http/impl/model/parser/ContentTypeHeader.scala

```diff
@@ -33,7 +34,7 @@ private[parser] trait ContentTypeHeader { this: Parser with CommonRules with Com
           case x: MediaType.WithOpenCharset if charset.isEmpty   => ContentType.WithMissingCharset(x)
         }
 
-      case Seq(("charset", value), tail @ _*) =>
+      case Seq((key, value), tail @ _*) if key.toRootLowerCase == "charset" =>
```

Comment:
`toRootLowerCase` takes code paths that take the locale into account (even though it's`Locale.ROOT`), so since we only want to know if it's equal to `"charset"` or not there's an opportunity for optimization here.

Hard to say if it's worth it here, but it feels like a relatively 'hot' part of the code? OTOH perhaps not compared to other handshaking/headerparsing.

Reply:
Great point. I didn't think too much about it. The reason I used it, is that there might be some potential for ambiguities where an attacker might use weird UTF-8 characters that might lowercase to ascii chars (but only in some servers). I guess we might want to have a fast case-insensitive ASCII char comparison.

Reply:
Now replaced with a custom matching method for ASCII chars.